[docs] Replace "See Also" sentences with proper blocks

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63506
* #63461

**Summary**
This commit replaces some plain sentences of the form "see also <link to
function documentation page> in `_torch_docs.py` with proper "See Also"
blocks supported by RsT.

**Test Plan**
Continuous integration.

Differential Revision: [D30799704](https://our.internmc.facebook.com/intern/diff/D30799704)

Commit:
[docs] Replace "See Also" sentences with proper blocks

**Summary**
This commit replaces some plain sentences of the form "see also <link to
function documentation page> in `_torch_docs.py` with proper "See Also"
blocks supported by RsT.

**Test Plan**
Continuous integration.

[ghstack-poisoned]

Commit:
Update on "[docs] Replace "See Also" sentences with proper blocks"


**Summary**
This commit replaces some plain sentences of the form "see also <link to
function documentation page> in `_torch_docs.py` with proper "See Also"
blocks supported by RsT.

**Test Plan**
Continuous integration.

[ghstack-poisoned]

Commit:
Update on "[docs] Replace "See Also" sentences with proper blocks"


**Summary**
This commit replaces some plain sentences of the form "see also <link to
function documentation page> in `_torch_docs.py` with proper "See Also"
blocks supported by RsT.

**Test Plan**
Continuous integration.

[ghstack-poisoned]

SplitInfinity commented on torch/_torch_docs.py:
```diff
@@ -9566,9 +9568,10 @@ def merge_dicts(*dicts):
-    This function is similar to NumPy's `take_along_axis`.
-    See also :func:`torch.gather`.
+.. seealso::
+        NumPy's `take_along_axis` which accepts an axis argument along which 1-D slices are taken.
```
Should I just delete this then? Remove it from the seealso?

mruberry replied:
It would actually be cool to call out the NumPy operators that are operations are based on but we don't have a template for it (yet) so yep, deleting this for now seems good.

SplitInfinity commented on torch/_torch_docs.py:
```diff
@@ -9566,9 +9568,10 @@ def merge_dicts(*dicts):
-    See also :func:`torch.gather`.
+.. seealso::
+        NumPy's `take_along_axis` which accepts an axis argument along which 1-D slices are taken.
+        :func:`torch.gather` which does not view the input as 1-D and accepts an additional argument
```
The name of the argument is `dim`. I will add that to the sentence.

mruberry replied:
take_along_dim also takes a dim argument -- I would just try to summarize the biggest user-facing difference

SplitInfinity replied:
The function of the `dim` argument in the case of `torch.gather` is clear from the docs, but not so for `torch.take_along_dim`. I tried playing with the operator myself:
```
>>> sorted_idx
tensor([[[0, 0],
         [1, 1]],

        [[0, 0],
         [1, 1]]])
>>> u
tensor([[[1, 2],
         [3, 4]],

        [[5, 6],
         [7, 8]]])
>>> torch.take_along_dim(u, sorted_idx, dim=1)
tensor([[[1, 2],
         [3, 4]],

        [[5, 6],
         [7, 8]]])
>>> torch.take_along_dim(u, sorted_idx, dim=2)
tensor([[[1, 1],
         [4, 4]],

        [[5, 5],
         [8, 8]]])
>>> torch.take_along_dim(u, sorted_idx)
tensor([1, 1, 2, 2, 1, 1, 2, 2])
```
and I don't quite understand how `dim=1` and `dim=2` work. I think it's just flattening first argument to `take_along_dim` when `dim` is not specified.

SplitInfinity replied:
Do you see what's going on? If not, can ask the core chat tomorrow.

SplitInfinity commented on torch/_torch_docs.py:
```diff
@@ -10496,11 +10499,9 @@ def merge_dicts(*dicts):
 
-``torch.where(condition)`` is identical to
-``torch.nonzero(condition, as_tuple=True)``.
+.. seealso::
```
Where should this text go? Should it replace the seealso box?

mruberry replied:
Yeah I would just make it the definition of this signature without putting it in a box

Commit:
Update on "[docs] Replace "See Also" sentences with proper blocks"


**Summary**
This commit replaces some plain sentences of the form "see also <link to
function documentation page> in `_torch_docs.py` with proper "See Also"
blocks supported by RsT.

**Test Plan**
Continuous integration.

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30799704).

Commit:
Update on "[docs] Replace "See Also" sentences with proper blocks"


**Summary**
This commit replaces some plain sentences of the form "see also <link to
function documentation page> in `_torch_docs.py` with proper "See Also"
blocks supported by RsT.

**Test Plan**
Continuous integration.

Differential Revision: [D30799704](https://our.internmc.facebook.com/intern/diff/D30799704)

[ghstack-poisoned]
